lib/virt_autotest/utils.pm

```diff
@@ -308,7 +309,7 @@ sub ensure_online {
     # Ensure guest is running
     # Only xen/kvm support to reboot guest at the moment
     if ($use_virsh && (is_xen_host || is_kvm_host)) {
-        if (script_run("virsh list | grep '$guest'") != 0) {
+        if (script_run("virsh domstate $guest | grep running") != 0) {
```

Comment:
I do not think this
```
script_run("virsh domstate $guest | grep running") != 0
```
is equal to
```
script_run("virsh list | grep '$guest'") != 0
```

Because "virsh list" list only active domains, including paused one, but "grep running" excludes paused domains. You can end up "starting paused domains". IMHO.